.gitbook/3-protocol-devs/1-dep-specs/4-data-validation.md

```diff
@@ -0,0 +1,217 @@
+| encrypted_data_base64 | string | Base64-encoded value after encrypt the original data             |
+| data_hash             | string | The value of hashing the original data with the SHA256 algorithm |
+
+#### Response Headers
+```
+# Content-Type: application/json
+```
+
+#### ResponseBody
+```json
+{
+  "unsigned_certificate": {
+    "cid": "QmeSiVLXWagUv9sLEHvbsUJy8rm7r5BoP2pAXrbx4pdbWi",
+    "unique_id": "9a3da3162aa592af3c77f1bba2d5635c7b4c065249bd36094fe3c11c73c90618",
+    "oracle_address": "panacea1ewugvs354xput6xydl5cd5tvkzcuymkejekwk3",
+    "deal_id": 1,
+    "provider_address": "panacea1ewugvs354xput6xydl5cd5tvkzcuymkejekwk3",
+    "data_hash": "13341f91f0da76d2adb67b519e2c9759822ceafd193bd26435ba0d5eee4c3a2b"
+  },
+  "signature": "MEQCIEPyGSe9wVIjrUFuzXtQtEc0siwaHkp4QJCMBvC8ttWQAiAUkntIQldgIkIBFdthaTRWXHisDV2Ys/Ufpc9zejUEuQ=="
+}
+```
+
+| Key                                   | Type   | Description                                                      |
+|---------------------------------------|--------|------------------------------------------------------------------|
+| unsigned_certificate                  | Object | Unsigned certificate containing data validation information      |
+| unsigned_certificate.cid              | string | A content identifier of IPFS                                     |
+| unsigned_certificate.unique_id        | string | UniqueID of the oracle that validated the data                   |
+| unsigned_certificate.oracle_address   | string | Account address of the oracle that validated the data            |
+| unsigned_certificate.deal_id          | int    | Deal to whom the provider intends to provide data                |
+| unsigned_certificate.provider_address | string | Data provider's account address                                  |
+| unsigned_certificate.data_hash        | string | The value of hashing the original data with the SHA256 algorithm |
+| signature                             | string | The value of unsigned_certificate signed with OracleKey          |
+
+
+### Data validation process
+
+#### Data Decryption
+Provider's encrypted data(`encrypted_data_base64`) can only be decrypted by Oracle.
+
+TODO: Guide to decrypt data
+```
+secret_key = SHA256(ECDH(oracle_private_key, provider_public_key))
+
+encrypted_data = Base64.Decode(encrypted_data_base64)
+
+orgin_data = AES256GCM.Decrypt(secret_key, encrypted_data)
+```
+
+#### Data Validation
+After hashing the original data with SHA256, it is encoded with HEX and compared with `data_hash`.
+```
+compare(data_hash, HEX.Encode(SHA256(origin_data))
+```
+
+Verify that the `provider_address` of the original data matches the JWT auth token issuer of the request header.
+```
+compare(origin_data.provider_address,jwtToken.issuer)
+```
+
+Deal is retrieved from the Panacea using the dealID got from the URI path.
+```
+deal = getDealFromPanacea(deal_id)
+```
```

Comment:
Here, readers won't understand what the URI path means since this chapter is way far from the API spec chapter. IMO, we can just simply say like `The deal information can be retrieved from Panacea using the deal ID`, since all information such as deal ID and encrypted data is from the HTTP POST request received from the user.

Reply:
You're right. I think the reader may be confused about what deal ID for path means.
I will fix it as your said.